test/integration/header_integration_test.cc

```diff
@@ -214,32 +214,31 @@ class HeaderIntegrationTest : public HttpIntegrationTest,
     initialize();
   }
 
-  virtual void createUpstreams() override {
+  void createUpstreams() override {
     HttpIntegrationTest::createUpstreams();
 
     if (use_eds_) {
       fake_upstreams_.emplace_back(new FakeUpstream(0, FakeHttpConnection::Type::HTTP2, version_));
     }
   }
 
-  virtual void initialize() override {
-    HttpIntegrationTest::initialize();
-
+  void initialize() override {
     if (use_eds_) {
-      eds_connection_ = fake_upstreams_[1]->waitForHttpConnection(*dispatcher_);
-      eds_stream_ = eds_connection_->waitForNewStream(*dispatcher_);
-      eds_stream_->startGrpcStream();
+      pre_worker_start_test_steps_ = [this]() {
```

Comment:
For my education, what caused this change to happen?

Reply:
I put a (buried) comment in the PR description:

> Note that the header integration test was relying on broken behavior
(EDS cluster would have no healthy hosts and fail, and then succeed on the
next iteration) so I had to fix the test after I streamlined the cluster manager code.

The header integration test only worked because of previously fault logic. Basically, the EDS cluster would initialize with no hosts, startup would attempt to make  request which would 503, then further init/callbacks would happen, and it would work on the next iteration. Because of all the cleanup that I did this is all fixed/consolidated, so this test was broken (correctly broken I would argue). So I had to fix the test to do EDS init before workers started.